x-pack/plugins/security/server/routes/authorization/roles/put.ts

```diff
@@ -62,26 +66,36 @@ export function definePutRolesRoutes({
       const { name } = request.params;
 
       try {
-        const { body: rawRoles } =
-          await context.core.elasticsearch.client.asCurrentUser.security.getRole(
+        const [features, { body: rawRoles }] = await Promise.all([
+          getFeatures(),
+          context.core.elasticsearch.client.asCurrentUser.security.getRole(
             { name: request.params.name },
             { ignore: [404] }
-          );
+          ),
+        ]);
+
+        const { validationErrors } = validateKibanaPrivileges(features, request.body.kibana);
```

Comment:
Since this validation requires the return value of the async `getFeatures`, I couldn't put this directly in the validation schema and had to implement it separately.